HumanPlayer.java

```diff
@@ -1,33 +1,24 @@
-import java.io.BufferedReader;
-
 public class HumanPlayer implements Player {
 
     private final char mark;
-    private final BufferedReader inputReader;
+    private final GameType gameType;
 
-    public HumanPlayer(BufferedReader inputReader, char mark) {
-        this.inputReader = inputReader;
+    public HumanPlayer(char mark, GameType gameType) {
         this.mark = mark;
+        this.gameType = gameType;
     }
 
     @Override
     public Integer nextMove(Board board) {
         int nextPosition = -1;
-        while (!validate(nextPosition)) {
-            try {
-                String line = inputReader.readLine();
-                nextPosition = Integer.parseInt(line);
-            } catch (Exception e) {
-                return nextPosition;
-            }
+        try {
+            nextPosition = Integer.parseInt(gameType.userInput());
```

Comment:
It is a bad name, but I cannot think of anything else and it tells them when its their turn.